packages/affiliates/libs/processors.js

```diff
@@ -147,9 +147,23 @@ function EmpBalances(handlers = {}, { collateral, tokens } = {}) {
     ContractExpired(caller) {
       // nothing
     },
+    // looking at the emp code, i think anyone can call this even if they never had a position
+    // this means balances may not exist or may go below 0. we should just catch those errors and ignore
     SettleExpiredPosition(caller, collateralReturned, tokensBurned) {
-      collateral.sub(caller, collateralReturned.toString());
-      tokens.sub(caller, tokensBurned.toString());
+      if (collateral.has(caller)) {
+        try {
+          collateral.sub(caller, collateralReturned.toString());
+        } catch (err) {
+          // ignore balance below 0 error
+        }
+      }
+      if (tokens.has(caller)) {
+        try {
+          tokens.sub(caller, tokensBurned.toString());
+        } catch (err) {
+          // ignore balance below 0 error
+        }
+      }
```

Comment:
convoluted logic, but when emps expire anyone without a position can settle. This was causing balances to go negative and throw errors. So now we just swallow the error, though maybe this could be done better.

Reply:
Hmmm, interesting case. Have you thought about querying only blocks _up to_ expiry? Have we decided whether to include expired contracts in developer mining?

Reply:
> Have you thought about querying only blocks up to expiry?

I have not.  Would be a bit more complicated to handle that though.

> Have we decided whether to include expired contracts in developer mining?

Theres been no decision around that, the calculation currently has no notion of an expired contract so it will continue to weight them for rewards.  That is something we should probably figure out though!